backend/api/Program.cs

```diff
@@ -10,6 +10,8 @@
 
 builder.Services.AddDbContext<FlotillaDbContext>(options => options.UseInMemoryDatabase("flotilla"));
 builder.Services.AddScoped<RobotService>();
+builder.Services.AddScoped<ReportService>();
+builder.Services.AddScoped<IsarService>(s => new IsarService(builder.Configuration.GetSection("Isar")["Url"]));
```

Comment:
```suggestion
builder.Services.AddScoped<ReportService>();
builder.Services.AddScoped<IIsarService, IsarService>(s => new IsarService(builder.Configuration.GetSection("Isar")["Url"]));
```
Since we have interfaces, this is how to add them to Dependency Injection.
All classes depending on these should require the interface, not the class, that is what makes it easily scalable in the future.

Reply:
Agreed